src/impl_float_maths.rs

```diff
@@ -0,0 +1,157 @@
+//! Element-wise methods for ndarray
+
+use num_traits::Float;
+
+use crate::imp_prelude::*;
+
+macro_rules! boolean_op {
+    ($($(#[$meta1:meta])* fn $id1:ident $(#[$meta2:meta])* fn $id2:ident -> $func:ident)+) => {
+        $($(#[$meta1])*
+        pub fn $id1(&self) -> Array<bool, D> {
+            self.mapv(A::$func)
+        }
+        $(#[$meta2])*
+        pub fn $id2(&self) -> bool {
+            self.mapv(A::$func).iter().any(|&b|b)
+        })+
+    };
+}
+
+macro_rules! unary_op {
+    ($($(#[$meta:meta])* fn $id:ident)+) => {
+        $($(#[$meta])*
+        pub fn $id(&self) -> Array<A, D> {
+            self.mapv(A::$id)
+        })+
+    };
+}
+
+macro_rules! binary_op {
+    ($($(#[$meta:meta])* fn $id:ident($ty:ty))+) => {
+        $($(#[$meta])*
+        pub fn $id(&self, rhs: $ty) -> Array<A, D> {
+            self.mapv(|v| A::$id(v, rhs))
+        })+
+    };
+}
+
+/// # Element-wise methods for Float Array
+///
+/// Element-wise math functions for any array type that contains float number.
+impl<A, S, D> ArrayBase<S, D>
+where
+    A: Float,
+    S: RawData<Elem = A> + Data,
+    D: Dimension,
+{
+    boolean_op! {
```

Comment:
If we add these methods, we should also consider adding `is_finite`, `is_any_finite`, and `is_all_finite`.

Reply:
In NumPy, the boolean arrays can be converted to a single `bool` value by `all` and `any` method.

```python
if (a > b).any():
    ...
```

Instead of the case of the [`Iterator::all`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.all) and the [`Iterator::any`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.any), would it be better to add `all` and `any` methods for boolean arrays?

Reply:
I'd prefer .all() or .any() to be implemented for all arrays - taking a closure, just like Iterator. Good separate PR. See comment in this PR for how to use existing `Zip::all`, i.e. the functionality already exists.